### PR TITLE
MINOR: Move Throttler to storage module

### PR DESCRIPTION
**Pull request by mimaison**
Throttler is one of the dependencies required in order to move LogCleaner out of core.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/log/LogCleaner.scala [hunk 1]
```diff
@@ -109,12 +110,12 @@ class LogCleaner(initialConfig: CleanerConfig,
...
   /* a throttle used to limit the I/O of all the cleaner threads to a user-specified maximum rate */
-  private[log] val throttler = new Throttler(desiredRatePerSec = config.maxIoBytesPerSecond,
-                                        checkIntervalMs = 300,
-                                        throttleDown = true,
-                                        "cleaner-io",
-                                        "bytes",
-                                        time = time)
+  private[log] val throttler = new Throttler(config.maxIoBytesPerSecond,
+                                             300,
+                                             true,
```
Comment (chia7712) [inline core/src/main/scala/kafka/log/LogCleaner.scala]: It seems `throttleDown` is always true in production. Maybe we can remove it?

**Review by chia7712 — COMMENTED** [review]
@mimaison nice patch! One small comment is left. Also, could you please cleanup the spotbugs file?

https://github.com/apache/kafka/blob/affe8da54c96d4038481c3807d059134961445a3/gradle/spotbugs-exclude.xml#L495

File: core/src/main/scala/kafka/log/LogCleaner.scala (hunk 1 above)
Comment (mimaison) [reply core/src/main/scala/kafka/log/LogCleaner.scala]: Right this can be removed. Done.

I also removed the spotbugs exclusion.

**Review by chia7712 — APPROVED** [review]
LGTM

**Comment by mimaison** [comment]
None of the test failures seem related, merging to trunk.

**Pull request merged**